expression/builtin_string.go

```diff
@@ -377,6 +401,14 @@ func (b *builtinConcatWSSig) evalString(row chunk.Row) (string, bool, error) {
 			sep = val
 			continue
 		}
+		targetLength += len(val)
+		if i > 1 {
+			targetLength += len(sep)
+		}
```

Comment:
Is it the time to move the `i==0` path out of this for loop?

Reply:
what do you mean by this? `i == 0` holds the separator.

Reply:
Consider this code style.
```go
N := len(args)
if N > 0 {
    // deal with seperator
}
for i := 1; i < N; i++ {
    // deal with strings
}
```

Otherwise it has too many if statement in this for loop